app/models/queen.rb

```diff
@@ -1,2 +1,25 @@
 class Queen < Piece
+  def move_legal?(to_row, to_col)
+    row_diff = (row - to_row).abs
+    col_diff = (col - to_col).abs
+    row_diff <= 1 && col_diff <= 1
```

Comment:
I don't think the expression (diff <= 1) is correct for a queen piece since it can move multiple squares in the three directions. It could potentially move vertically the entire height of the board to capture the opposing queen.